src/lib.rs

```diff
@@ -1,5 +1,11 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 #![cfg_attr(target_arch = "xtensa", feature(asm_experimental_arch))]
+#![cfg_attr(
```

Comment:
Nothing to do here but just noticing: using incomplete features is oh so not nice, but oh well. They rushed into using `async-fn-in-trait` in `embedded-hal-async`.

Reply:
I 100% agree.

The fact that they give mutable self without a lifetime is also frustrating - they actually had it that way originally, but removed it.

So much of this feels like treading on fresh snow, and it's not obvious why - like we don't have to literally make everything unstable all at once.
Anyway.